cli/dcoscli/config/main.py

```diff
@@ -86,28 +85,18 @@ def _set(name, value):
     """
 
     if name == "core.dcos_url":
-        return _cluster_setup(value)
+        notice = (
+            "This config property is being deprecated. "
+            "To setup the CLI to talk to your cluster, please run "
+            "`dcos cluster setup <dcos_url>`.")
+        emitter.publish(DefaultError(notice))
```

Comment:
Based on our offline discussion, here's the semantics I think we want to have / preserve:
```
Setup flow for new users of the 0.4.x CLI:
$ export DCOS_CONFIG=<path/to/config> (optional)
   (default when not set is ~/.dcos/dcos.toml)
$ dcos config set core.dcos_url <url>
$ dcos config set core.ssl_verify [“true” | “false” | <path/to/ca_cert>]
$ dcos auth login
   (generates  / updates global config at $DCOS_CONFIG where all settings are set)

Setup flow for new users of the 0.5.x CLI:
$ export DCOS_DIR=<path/to/config_dir> (optional)
   (default when not set is ~/.dcos)
$ dcos cluster setup <url>
   (generates / updates per cluster config under $DCOS_DIR/clusters/<cluster_id> where all settings are set)
   (sets newly set up cluster config as the active one)

Automatic update of config when currently using the 0.4.x CLI and updating to new 0.5.x CLI:
$ export DCOS_CONFIG=<path/to/config> (optional)
  (default when not set is ~/.dcos/dcos.toml)
$ dcos <subcommand>

$ … download and install new dcos 0.5.x binary …

$ export DCOS_DIR=<path/to/config_dir> (optional)
   (default when not set is ~/.dcos)
$ dcos <subcommand>
   (triggers conversion from old config structure to new)
   (moves global config from $DCOS_CONFIG to $DCOS_DIR/clusters/<cluster_id>/dcos.toml)
   (sets newly converted cluster config as the active one)
   (DCOS_CONFIG is no longer honored)

Preserving backwards compatibility:
1. Before `dcos cluster setup <url>` is called, follow the old flow semantics
    and allow users to change the config pointed to by $DCOS_CONFIG
    (print warning message saying these calls are deprecated and recommend using `dcos cluster setup`)
2 .After `dcos cluster setup` is called (or automatic conversion has occurred),
    update the ‘active’ cluster’s config whenever old flow commands are called
    (print different warning message saying the calls are deprecated and cluster config state may now be corrupted)
```